deployment/cfn/application.py

```diff
@@ -270,29 +299,75 @@ def create_load_balancer(self, app_server_lb_security_group):
 
     def create_auto_scaling_resources(self, app_server_security_group,
                                       app_server_lb):
-        app_server_launch_config_name = 'lcAppServer'
+        self.add_condition('BlueCondition', Equals('Blue', Ref(self.color)))
+        self.add_condition('GreenCondition', Equals('Green', Ref(self.color)))
 
-        app_server_launch_config = self.add_resource(
+        blue_app_server_launch_config = self.add_resource(
             asg.LaunchConfiguration(
-                app_server_launch_config_name,
+                'lcAppServerBlue',
+                Condition='BlueCondition',
                 ImageId=Ref(self.app_server_ami),
                 IamInstanceProfile=Ref(self.app_server_instance_profile),
                 InstanceType=Ref(self.app_server_instance_type),
                 KeyName=Ref(self.keyname),
-                SecurityGroups=[Ref(app_server_security_group)]
+                SecurityGroups=[Ref(app_server_security_group)],
+                UserData=Base64(
```

Comment:
User data can be shell scripts, but it can also be [`cloud-init`](http://cloudinit.readthedocs.org/en/latest/index.html) directives. Some more AWS specific details can be found [here](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html#user-data-cloud-init).

Unfortunately, `cloud-init` has a history of never really being well defined, and each distribution runs a different version. That said, I still like it better than arbitrary shell scripts. Also, CoreOS has a nice [validator](https://coreos.com/validate/), but it only really works with its flavor of `cloud-init`.